src/ceph_osd.cc

```diff
@@ -472,6 +473,10 @@ int main(int argc, const char **argv)
     forker.exit(0);
   }
   
+  if (extblkdev::preload(g_ceph_context) < 0) {
+    forker.exit(1);
+  }
+
```

Comment:
@mohmacht I really think the problem is with this piece of code. All of the teuthology tests are failing with `Command failed on smithi112 with status 1: 'sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage daemon-helper kill ceph-osd -f --cluster ceph -i 0' `, which means that when the OSD is getting restarted, this piece of code is getting called and causing problems.

I managed to reproduce the issue with your branch on a vstart cluster by running `./bin/init-ceph restart osd.0`. You can see "Got signal terminated" in the osd log when that command is run.

I am also emailing you a teuthology.log and an osd log from one of the failed tests so you can inspect them yourself.

I'd also encourage you to request sepia access so you can view/debug teuthology test results. You can do that by following the procedure here: https://ceph.github.io/sepia/adding_users/

But the logs I'm emailing you are really all that teuthology collects.

Reply:
Laura, thanks for checking out the fails. I have rebuilt the ceph ebdplugin branch on a RHEL8 VM, started ceph with `../src/vstart.sh --new -x --localhost` and ran `./bin/init-ceph restart` to shutdown the OSD. I did so without any failure... :-(. Maybe my patch has a problem when applied on the current main branch, I will rebase to the current main and try again.

Reply:
Even with my patch applied to the most recent commit on the main branch, the OSD shuts down and restarts cleanly.
Maybe there are differences in our boundary condition? I am starting under my own account, not root, and there is no ceph user existing on the machine. The OSD uses bluestone and is backed by files in the build/dev/osd* directory. I am manually launching `../src/vstart.sh --new -x --localhost --bluestore` followed by  `/bin/init-ceph restart osd.0`, with a few seconds between the two commands. The out/osd.0.log shows a clean shutdown:
```2022-09-13T15:21:41.587-0400 7ff067b29640  1 bdev(0x5581cd22c300 /home/mohmacht/ceph/build/dev/osd0/block) close
2022-09-13T15:21:41.831-0400 7ff067b29640  1 freelist shutdown
2022-09-13T15:21:41.831-0400 7ff067b29640  1 bdev(0x5581cd22be00 /home/mohmacht/ceph/build/dev/osd0/block) close
2022-09-13T15:21:42.203-0400 7ff067b29640  0 osd.0 0 Slow Shutdown duration:1.433048 seconds
```
This last ran was done on a VM running ubuntu 20.4. Are there any other parameters that may different between our runs?

Reply:
I am building ceph with `run-make-check.sh`. Are there other build parameters that I should use to get closer to the setup that failed?

Reply:
@mohmacht, I apologize -- I rechecked my "reproducer" and found that the same "Got signal terminated" message in the OSD log also occurs on the main branch. This must be a normal symptom on the vstart cluster when restarting an OSD.

I'm trying out a few more methods locally to see if I can accurately recreate the issue.

Reply:
Somehow, we're running into a situation where the cluster has 3 OSDs in, but when it tries to bring them up, there is a failure.
```
1662760631.0800695 mon.a (mon.0) 9 : cluster [DBG] osdmap e2: 0 total, 0 up, 0 in
1662760631.371903 mon.a (mon.0) 12 : cluster [DBG] osdmap e3: 1 total, 0 up, 1 in
1662760631.574212 mon.a (mon.0) 15 : cluster [DBG] osdmap e4: 2 total, 0 up, 2 in
1662760631.774574 mon.a (mon.0) 18 : cluster [DBG] osdmap e5: 3 total, 0 up, 3 in
<ends after this point>
```
A successful test brings the ODSs up:
```
1662945178.5975142 mon.a (mon.0) 9 : cluster [DBG] osdmap e2: 0 total, 0 up, 0 in
1662945178.9491317 mon.a (mon.0) 12 : cluster [DBG] osdmap e3: 1 total, 0 up, 1 in
1662945179.1566288 mon.a (mon.0) 15 : cluster [DBG] osdmap e4: 2 total, 0 up, 2 in
1662945179.3707914 mon.a (mon.0) 18 : cluster [DBG] osdmap e5: 3 total, 0 up, 3 in
1662945181.5337565 mon.a (mon.0) 23 : cluster [DBG] osdmap e6: 3 total, 0 up, 3 in
1662945182.5393124 mon.a (mon.0) 30 : cluster [DBG] osdmap e7: 3 total, 0 up, 3 in
1662945183.537131 mon.a (mon.0) 36 : cluster [INF] osd.1 [v2:172.21.15.57:6800/15832,v1:172.21.15.57:6801/15832] boot
1662945183.5371602 mon.a (mon.0) 37 : cluster [DBG] osdmap e8: 3 total, 1 up, 3 in
```